lldb/test/API/functionalities/fork/concurrent_vfork/main.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <mutex>
+#include <sys/wait.h>
+#include <thread>
+#include <unistd.h>
+#include <vector>
+
+pid_t g_pid = 0;
+std::mutex g_child_pids_mutex;
+std::vector<pid_t> g_child_pids;
+
+int call_vfork(int index) {
```

Comment:
We probably want to test with both `fork()` and `vfork()` here and we want to know if we want to call `exec*()` or not. If we add a `const char *exec_path` to the call and this is not NULL, then we call exec. So maybe change this to be:
```
int call_vfork(int index, bool use_vfork, const char *exec_path)
```